packages/loopring_v3/contracts/test/AmmPool.sol

```diff
@@ -360,111 +400,340 @@ contract AmmPool is IBlockReceiver {
-            delete queue[n];
+            ctx.ammBalances[i] = ctx.ammBalances[i].add(amount);
         }
-        queuePos += auxData.numItems;
 
-        // Deposit/Withdraw to/from the AMM account when necessary
-        for (uint i = 0; i < tokens.length; i++) {
-            if (ammBalances[i] > ammBalancesInAccount[i]) {
-                uint amount = ammBalances[i] - ammBalancesInAccount[i];
-                // Check that the deposit in the block matches the expected deposit
-                DepositTransaction.Deposit memory _deposit = _block.readDeposit(txIdx++);
-                require(_deposit.owner == address(this), "INVALID_TX_DATA");
-                require(_deposit.accountID == accountID, "INVALID_TX_DATA");
-                require(_deposit.tokenID == tokens[i].tokenID, "INVALID_TX_DATA");
-                require(_deposit.amount == amount, "INVALID_TX_DATA");
-                // Now do this deposit
-                uint ethValue = 0;
-                if (tokens[i].addr == address(0)) {
-                    ethValue = _deposit.amount;
-                } else {
-                    // Approve the deposit transfer
-                    ERC20(tokens[i].addr).approve(address(exchange), _deposit.amount);
-                }
-                exchange.deposit{value: ethValue}(
-                    _deposit.owner,
-                    _deposit.owner,
-                    tokens[i].addr,
-                    uint96(_deposit.amount),
-                    new bytes(0)
-                );
-                numTransactionsConsumed++;
-                // Total balance in this contract decreases by the amount deposited
-                totalBalance[tokens[i].addr] = totalBalance[tokens[i].addr].sub(amount);
-            } else if (ammBalancesInAccount[i] > ammBalances[i]) {
-                uint amount = ammBalancesInAccount[i] - ammBalances[i];
-                // Check that the withdrawal in the block matches the expected withdrawal
-                WithdrawTransaction.Withdrawal memory withdrawal = _block.readWithdrawal(txIdx++);
-                require(withdrawal.owner == address(this), "INVALID_TX_DATA");
-                require(withdrawal.accountID == accountID, "INVALID_TX_DATA");
-                require(withdrawal.tokenID == tokens[i].tokenID, "INVALID_TX_DATA");
-                require(withdrawal.amount == amount, "INVALID_TX_DATA");
-                require(withdrawal.feeTokenID == withdrawal.tokenID, "INVALID_TX_DATA");
-                require(withdrawal.fee == 0, "INVALID_TX_DATA");
-                require(withdrawal.to == address(this), "INVALID_TX_DATA");
-                require(withdrawal.extraData.length == 0, "INVALID_TX_DATA");
-                // Now approve this withdrawal
-                withdrawal.validUntil = 0xffffffff;
-                bytes32 txHash = WithdrawTransaction.hashTx(exchangeDomainSeparator, withdrawal);
+        // Mint liquidity tokens
+        mint(join.owner, join.poolAmountOut);
```

Comment:
Hmm... I thought this `join.poolAmountOut` should be calculated instead of provided by the user in the `joinPool` function... same for `poolAmountIn`

Reply:
I just stuck with how balancer does things here, not necessarily the best way to do things.